fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/search/FhirUriParameterDefinition.java

```diff
@@ -155,6 +178,60 @@ public void isValidModifier(SearchRequestParameterModifier modifier) {
 		if (!type.availableModifiers.contains(modifier)) {
 			throw FhirException.createFhirError(String.format("Invalid modifier '%s' for parameter [%s:%s].", modifier, name, type.name()), OperationOutcomeCode.MSG_PARAM_INVALID);
 		}
+	}
+	
+	public void validateValues(final Collection<PrefixedValue> prefixedValues) {
+		
+		if (!isMultipleValuesSupported() && prefixedValues.size() > 1) {
+			String values = Arrays.toString(prefixedValues.stream().map(PrefixedValue::getValue).collect(Collectors.toSet()).toArray());
+			throw FhirException.createFhirError(String.format("Too many filter parameter values %s are submitted for parameter '%s'.", values, getName()), OperationOutcomeCode.MSG_PARAM_INVALID, SEARCH_REQUEST_PARAMETER_MARKER);
+		}
+		
+		Set<String> supportedValues = getSupportedValues();
+		Set<String> values = prefixedValues.stream().map(PrefixedValue::getValue).collect(Collectors.toSet());
+		
+		Set<String> uppercaseValues = values.stream()
+				.map(String::toUpperCase)
+				.collect(Collectors.toSet());
+		
+		if (!supportedValues.isEmpty() && !supportedValues.containsAll(uppercaseValues)) {
+			throw FhirException.createFhirError(String.format("Filter parameter value %s is not supported. Supported parameter values are %s.", Arrays.toString(values.toArray()), Arrays.toString(supportedValues.toArray())), 
+				OperationOutcomeCode.MSG_PARAM_UNKNOWN, "SEARCH_REQUEST_PARAMETER_MARKER");
+		}
+		
+		for (String value : values) {
+			switch (type) {
+			case NUMBER:
+				try {
```

Comment:
How about a single big try-catch block around the entire switch case statement instead of 4 small ones?